### ISSUE TITLE
How to import a case

### ISSUE THREAD

**Issue by doantranthanh** [o19s/quepid#227]
hi all,

I am trying to import an exported case in my local quepid. 

For example, I exported a case from our Quepid system where the content experts ranked the document. I want to import it to my local for my own testing before actually put it back to the Quepid system.

However I am not able to do that. Is there any way I can do it?

Many thanks

Thanh

**Comment by epugh** [comment]
Hi @doantranthanh can you share a screnshot and a sample of the document you are importing?    Also, go into developer console and check what you got back form the javascript layer.  

I'm going to give it a test now myself....

**Comment by doantranthanh** [comment]
hi @epugh 

Please have a look at the images for your reference

I started import my exported case

![image](https://user-images.githubusercontent.com/9557564/96972381-5aadd800-150e-11eb-84ae-0cb8f1e6da35.png)

Here is the error. It looks like the case is not allowed to import due to the mismatch header

![image](https://user-images.githubusercontent.com/9557564/96972429-6a2d2100-150e-11eb-8ac9-3351f5ff5952.png)

Here is the developer console log

![image](https://user-images.githubusercontent.com/9557564/96972607-9c3e8300-150e-11eb-9a2d-67eecf6732b4.png)

And finally, here is my exported case 

````
Team Name,Case Name,Case ID,Query Text,Score,Date Last Scored,Count,Notes
"Polaris dev team, US Editors",SP-001- DevGB-quepid test,27,paternity,0,2020-10-21T11:09:22.000Z,848,
"Polaris dev team, US Editors",SP-001- DevGB-quepid test,27,Maternity,0,2020-10-21T11:09:22.000Z,1133,
"Polaris dev team, US Editors",SP-001- DevGB-quepid test,27,on annual leave,0.35728227265773876,2020-10-21T11:09:22.000Z,5378,
"Polaris dev team, US Editors",SP-001- DevGB-quepid test,27,""on annual leave"",0,2020-10-21T11:09:22.000Z,5378,
"Polaris dev team, US Editors",SP-001- DevGB-quepid test,27,human rights,0,2020-10-21T11:09:22.000Z,3531,
````

Also, when I tried to import the following snapshot

````
Snapshot Name,Snapshot Time,Case ID,Query Text,Doc ID,Doc Position
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,paternity,113063,1
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,paternity,113062,2
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,paternity,87372,3
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,paternity,155311,4
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,paternity,49611,5
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,paternity,111642,6
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,paternity,153132,7
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,paternity,28908,8
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,paternity,153133,9
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,paternity,150317,10
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,Maternity,113052,1
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,Maternity,113059,2
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,Maternity,7132,3
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,Maternity,57414,4
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,Maternity,153138,5
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,Maternity,64934,6
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,Maternity,7076,7
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,Maternity,7112,8
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,Maternity,7221,9
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,Maternity,7474,10
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,on annual leave,39684,1
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,on annual leave,150460,2
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,on annual leave,165128,3
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,on annual leave,92305,4
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,on annual leave,113014,5
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,on annual leave,156128,6
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,on annual leave,58150,7
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,on annual leave,112202,8
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,on annual leave,95975,9
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,on annual leave,98210,10
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,""on annual leave"",156128,1
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,""on annual leave"",158833,2
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,""on annual leave"",155925,3
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,""on annual leave"",165316,4
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,""on annual leave"",103546,5
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,""on annual leave"",94128,6
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,""on annual leave"",112017,7
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,""on annual leave"",112022,8
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,""on annual leave"",163556,9
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,""on annual leave"",163557,10
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,human rights,20368,1
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,human rights,158653,2
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,human rights,150045,3
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,human rights,150207,4
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,human rights,51205,5
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,human rights,111499,6
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,human rights,150208,7
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,human rights,63907,8
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,human rights,66482,9
Snapshot: 21/10/20,2020-10-21T10:20:09.000Z,27,human rights,152024,10
````
It is said that

![image](https://user-images.githubusercontent.com/9557564/96972927-14a54400-150f-11eb-8d83-562dba52cda8.png)

![image](https://user-images.githubusercontent.com/9557564/96972973-2a1a6e00-150f-11eb-878f-2f2c200dda62.png)

I thought because I did not import the case first then I am not be able to import the rating but I might be wrong.

Also, sometimes when I shared my cases to my colleagues then they could not see the updated rank score that were changed by me. Sometimes we need to wait for few hours to see the updates. I have no idea why. 

Please could you have a look and support me?

Many thanks.

**Comment by epugh** [comment]
Thanks for a really great ticket!  Sorry I didn't get to testing, but I will today or tomorrow!

**Comment by epugh** [comment]
Let me see if I understand what you are trying to do.  You have a case set up in Quepid, with queries like `paternity` and `Maternity`.    You export that case...    Then, are you trying to import it into a different Quepid instance?   Or, are you then having it be rated externally, and so what you really want is to import the new ratings that were done internally?  

Looking at the screenshots, your import of CSV is failing because the Import Snapshot has different headings than what your CSV file has.  

In the screenshot below, do you see the "Basic"...   It can be more useful for moving around ratings.  `General`, `Detailed`, and `Snapshot` are kind of legacy ones, that drag a lot of extra data along, that you may not actually need.  

![Screenshot at Oct 26 17-52-24](https://user-images.githubusercontent.com/22395/97232718-12174880-17b4-11eb-94b2-9af629b08f65.png)

Let me know if this helps, or if we need to jump on a Zoom!

**Comment by epugh** [comment]
I wanted to check and see if your issue has been resolved?

**Comment by doantranthanh** [comment]
Hi @epugh 

Yeah I tried to import an exported case from a Quepid instance to another Quepid instance. In particularly, the exported case is created by our content experts and BA in our Quepid system so I would like to import it to my local so I can test my tuning query before checking in/present in the Quepid system.

I tried to export the "Basic" case as you guided but there is no luck for importing to my Quepid instance. It looks like this

````
query,docid,rating
paternity,113063,1
paternity,113062,1
paternity,87372,1
paternity,155311,1
paternity,49611,
paternity,111642,
paternity,153132,1
````
However, it is not working. Is this because the ElasticSearch instance on my local is different from our ES system. I dont think so because they are having the same settings, mappings and data.

Cheers

Thanh

**Comment by epugh** [comment]
@doantranthanh have you seen the fix I just committed #252 ?  I think that previously you might have been getting a validation error on the front end for the line `paternity,49611,` and `paternity,111642,
`, since it didn't have a rating, however now that is supported.  we would import the query, and just rating combo, but you would still get the query.

I tried it locally and they imported.

![Screenshot at Nov 21 10-13-21](https://user-images.githubusercontent.com/22395/99880572-38c57500-2be2-11eb-85b5-9d1a10e8cb63.png)
